src/server/world.zig

```diff
@@ -761,23 +761,70 @@ pub const ServerWorld = struct { // MARK: ServerWorld
 		self.itemDropManager.loadFrom(zon);
 	}
 
-
 	pub fn findPlayer(self: *ServerWorld, user: *User) void {
-		var buf: [1024]u8 = undefined;
-		const playerData = files.readToZon(main.stackAllocator, std.fmt.bufPrint(&buf, "saves/{s}/player/{s}.zig.zon", .{self.name, user.name}) catch "") catch .null; // TODO: Utils.escapeFolderName(user.name)
+		var dest: [1024]u8 = undefined;
+		const hashedName = std.base64.url_safe.Encoder.encode(&dest, user.name);
+
+		var buf: [32768]u8 = undefined;
+		const playerData = files.readToZon(main.stackAllocator, std.fmt.bufPrint(&buf, "saves/{s}/players/{s}.zig.zon", .{self.name, hashedName}) catch "") catch .null; // TODO: Utils.escapeFolderName(user.name)
 		defer playerData.deinit(main.stackAllocator);
 		const player = &user.player;
 		if(playerData == .null) {
-			// Generate a new player:
 			player.pos = @floatFromInt(self.spawn);
 		} else {
-			player.loadFrom(playerData);
+			player.loadFrom(playerData.getChild("entity"));
+
+			main.items.Inventory.Sync.setGamemode(user, std.meta.stringToEnum(main.game.Gamemode, playerData.get([]const u8, "gamemode", "survival")).?);
+		}
+	}
+
+	pub fn savePlayer(self: *ServerWorld, user: *User) !void {
+		var dest: [1024]u8 = undefined;
+		const hashedName = std.base64.url_safe.Encoder.encode(&dest, user.name);
+
+		var buf: [32768]u8 = undefined;
+		const path = try std.fmt.bufPrint(&buf, "saves/{s}/players/{s}.zig.zon", .{self.name, hashedName});
+		
+		var playerZon: ZonElement = files.readToZon(main.stackAllocator, path) catch ZonElement.initObject(main.stackAllocator);
```

Comment:
As mentioned on discord, readToZon might return an array, instead of an object.
This will result in a crash when calling put. Please check if playerZon is an object, and if not create a new one.